File: python/ip.py
import socket

# Good enough for my test project. Check https://stackoverflow.com/questions/319279/how-to-validate-ip-address-in-python
# if you want to go further.
def isValidIP(ip):
    ad = ip.split(":")
    if len(ad) > 2:
        return False
    try:
        socket.inet_aton(ad[0])
    except socket.error:
        return False
    if len(ad) > 1:
        # ":" not followed by a port number?
        if len(ad[1])>0:
            try:
                _ = int(ad[1])
                return True
            except:
                return False
    # IP address without a port number.
    return False

File: node/lib/ip.js
const checkIp = require('check-ip')

const isValidIP = ip => {
  const ad = ip.split(':') // checkIp does not handle ports...
  if (ad.length > 2) {
    return false
  }
  const host = ad[0] === 'localhost' ? '127.0.0.1' : ad[0]
  if (!checkIp(host).isValid) {
    return false
  }
  if (ad.length > 1) {
    // ':' not followed by a port number?
    if (ad[1].length > 0) {
      const port = parseInt(ad[1])
      if (isNaN(port)) {
        return false
      }
      return true
    }
  }
  // IP address without a port number.
  return false
}

module.exports = isValidIP
